packages/server/src/api/muster/muster.controller.ts

```diff
@@ -231,30 +233,76 @@ class MusterController {
 
   /**
    * This method returns the normalized rate (0 - 1.0) of compliance on a given
-   * date for a given unit, against  of the unit's muster configs.
+   * date for a given unit, against all of the unit's muster configs.
    */
   async getMusterComplianceByDate(
     req: ApiRequest<{ orgId: number; unitName: string }, null, GetMusterComplianceByDateQuery>,
     res: Response<GetMusterComplianceByDateResponse>,
   ) {
-    const outValue = { musterComplianceRate: 0 };
-    const reportDate = dateFromString(req.query.timestamp);
-    if (!reportDate) {
-      throw new BadRequestError('Missing reportDate.');
+    const outValue = { musterComplianceRate: 0, isoDate: req.query.isoDate };
+    outValue.musterComplianceRate = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, req.query.isoDate);
+    res.json(outValue);
+  }
+
+  /**
+   * This method returns the normalized rate (0 - 1.0) of compliance on a given
+   * date range for a given unit, against all of the unit's muster configs.
+   */
+  async getMusterComplianceByDateRange(
+    req: ApiRequest<{ orgId: number; unitName: string }, null, GetMusterComplianceByDateRangeQuery>,
+    res: Response<GetMusterComplianceByDateRangeResponse>,
+  ) {
+    const out: GetMusterComplianceByDateRangeResponse = { musterComplianceRates: [] };
+    const range: string[] = req.query.isoDateRange.split('/');
+    if (range.length !== 2) {
+      throw new BadRequestError('Invalid ISO date range.');
+    } else if (range[0] === range[1]) {
+      const compliance = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, range[0]);
+      out.musterComplianceRates.push({ musterComplianceRate: compliance, isoDate: range[0] });
+    } else {
+      let fromDate: moment.Moment = moment(range[0]);
+      const toDate: moment.Moment = moment(range[1]);
+      while (fromDate <= toDate) {
+        const compliance = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, fromDate.toISOString());
+        out.musterComplianceRates.push({ musterComplianceRate: compliance, isoDate: fromDate.toISOString() });
+        fromDate = fromDate.add(1, 'day');
+      }
     }
+    res.json(out);
+  }
+
+}
+
+/**
+ * This helper method returns the normalized rate (0 - 1.0) of compliance on a given
+ * date for a given unit, against all of the unit's muster configs.
+ * @param orgId - the organization id
+ * @param unitName - the roster the users belong to
+ * @param isoDate - the date in format YYYY-MM-DD
+ */
+async function getMusterComplianceByDate(orgId: number, unitName: string, isoDate: string) {
```

Comment:
This function is fairly long making it harder to read given the size (every single line needs to be read). When I see comments like this...
// handle one-time muster
// handle recurring muster...
...typically it indicates areas of code which can be extracted out as functions.

Reply:
The **if** and **else** blocks represent about 2/3 of the code in this method. Both code blocks can be extracted out to functions. When extracted, it helps with keeping the scope of local variables small, the scope not extending to all parts of the large function.